### PR TITLE
Define storage for saving info about legacy chain brackets - Closes #7545

### LINKED ISSUE #7545: Define storage for saving info about legacy chain brackets
### Description

- Create a new `DB_KEY_LEGACY_BRACKETS` for storing legacy chain brackets
- Each bracket is identified by `Buffer.concat([DB_KEY_LEGACY_BRACKETS, bracket.snapshotBlockID])` with value as 
```ts
interface LegacyChainBracketInfo { 
  startHeight: number;
  snapshotBlockHeight: number;
  lastBlockHeight: number;
}
```
- In `LegacyChainHandler.init`, if there is no key exists for a `snapshotBlockID` we initialize as below,

**key:** 
```ts
config.bracket[index].snapshotBlockID
```
**value:**  
```ts
{
  startHeight: config.bracket[index].startHeight,
  snapshotBlockHeight: config.bracket[index].snapshotBlockHeight,
  lastBlockHeight:  config.bracket[index].snapshotBlockHeight
}
```

- Implement `getLegacyChainBracketInfo(snapshotBlockID): Promise<LegacyChainBracket>`
- Implement `setLegacyChainBracketInfo(snapshotBlockID: Buffer, bracketInfo: LegacyChainBracket): Promise<void>` 

### Acceptance Criteria

- Should have unit test corresponding to it

### PR DESCRIPTION
**Pull request by mosmartin**
### What was the problem?

This PR resolves #7545

### How was it solved?

- Update codecs
- Update the legacy schema
- Add getLegacyChainBracketInfo() method
- Add setLegacyChainBracketInfo() method

### How was it tested?

Added unit tests

### CODE REVIEW

File: framework/src/engine/legacy/legacy_chain_handler.ts [hunk 1]
```diff
@@ -31,9 +32,19 @@ export class LegacyChainHandler {
 		this._legacyConfig = args.legacyConfig;
 	}
 
-	// eslint-disable-next-line @typescript-eslint/require-await
 	public async init(args: LegacyHandlerInitArgs): Promise<void> {
+		const storage = new Storage(this._db as Database);
 		this._db = args.db;
+
+		for (const bracket of this._legacyConfig.brackets) {
+			if (!bracket.snapshotBlockID) {
```
Comment (ishantiw) [inline framework/src/engine/legacy/legacy_chain_handler.ts:40]: you also need to check `storage.getLegacyChainBracketInfo(bracket.snapshotBlockID)` exists or not, we only specify defaults when it doesn't exist in DB

File: framework/src/engine/legacy/schemas.ts [hunk 2]
```diff
@@ -80,3 +80,14 @@ export const blockHeaderSchemaV2 = {
 		'asset',
 	],
 };
+
+export const legacyChainBracketInfoSchema = {
+	$id: '/lisk/legacyChainBracketInfo',
```
Comment (ishantiw) [inline framework/src/engine/legacy/schemas.ts:85]: Maybe?
```suggestion
	$id: '/legacy/legacyChainBracketInfo',
```

**Review by ishantiw — CHANGES_REQUESTED** [review]
LGTM, one comment

**Review by ishantiw — APPROVED** [review]
Well done 👏

File: framework/test/unit/engine/legacy/storage.spec.ts [hunk 3]
```diff
@@ -131,4 +131,57 @@ describe('Legacy storage', () => {
...
+
+			await storage.setLegacyChainBracketInfo(header.id, bracketInfo);
+
+			await expect(storage.getLegacyChainBracketInfo(Buffer.alloc(0))).rejects.toThrow(
+				`Specified key 02 does not exist`,
+			);
+		});
+	});
+
+	describe('getLegacyChainBracketInfo', () => {
```
[the remark this answers is not in the document]
Comment (mosmartin) [reply framework/test/unit/engine/legacy/storage.spec.ts:135]: In my opinion it's self explanatory, no need for comments.

File: framework/test/unit/engine/legacy/storage.spec.ts [hunk 4]
```diff
@@ -131,4 +131,27 @@ describe('Legacy storage', () => {
...
+				startHeight: header.height,
+				snapshotBlockHeight: header.height,
+				lastBlockHeight: header.height,
+			};
+
+			await storage.setLegacyChainBracketInfo(header.id, bracketInfo);
+
+			const result = await storage.getLegacyChainBracketInfo(header.id);
+
+			expect(result).toEqual(encodeLegacyChainBracketInfo(bracketInfo));
```
[the remark this answers is not in the document]
Comment (mosmartin) [reply framework/test/unit/engine/legacy/storage.spec.ts:148]: For readability!

**Pull request merged**